src/Kuzzle.js

```diff
@@ -407,7 +407,10 @@ class Kuzzle extends KuzzleEventEmitter {
     }
 
     // we follow the api but allow some more logical "mistakes" (the only allowed value for refresh arg is "wait_for")
-    if (request.refresh) {
+    if (request.refresh || options.refresh) {
+      if (options.refresh) {
+        delete options.refresh;
```

Comment:
Why do you delete the property after using it? 
It can be confusing to mutate the original object, it could be reused for subsequent calls:
```
const options = { refresh: 'wait_for' };
sdk.query({...}, options);
sdk.query({...}, options); // options object have been mutated 
```